packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -259,6 +259,7 @@ class ImageCache {
         });
       }
       _currentSizeBytes -= image.sizeBytes!;
+      image.completer.keepAlive = false;
```

Comment:
Could somebody else have gained access to the completer and set `keepAlive` to true? We would be overwriting their request to keep the thing alive here. Does completer instead need to have a keepAlive ref count that gets decreased/increased and its only no longer kept alive when it reaches zero?

Reply:
I was hoping to avoid adding a new kind of refcount parameter, but maybe that would be better here rather than relying on the listener list to be the refcount. That might resolve your other concern below as well.

Reply:
I am not saying that a ref count is the right idea here. Just worried that just having a boolean leads to other problems...